lib/src/screens/home_screen/home_screen.dart

```diff
@@ -1,12 +1,114 @@
+import '../../blocs/coins_bloc/coins_bloc.dart';
+import '../../blocs/coins_bloc/coins_state.dart';
+import '../../constants/colors.dart';
+import '../../constants/strings.dart';
+import '../../constants/text_style.dart';
+import '../detail_screen/detail_screen.dart';
+import '../../widgets/coin_card.dart';
+import '../../widgets/search_bar.dart';
 import 'package:flutter/material.dart';
-class HomePage extends StatefulWidget
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+class HomeScreen extends StatefulWidget {
   @override
-  _HomePageState createState() => _HomePageState();
+  _HomeScreenState createState() => _HomeScreenState();
 }
 
-class _HomePageState extends State<HomePage> {
+class _HomeScreenState extends State<HomeScreen> {
   @override
   Widget build(BuildContext context) {
-    return Container();
+    return Scaffold(
+      backgroundColor: ColorsApp.backgroundColor,
+      body: SafeArea(
+        child: Column(
+          children: [
+            Container(
+              height: 200,
+              child: Column(
+                children: [
+                  BlocBuilder<CoinsBloc, CoinsState>(builder: (context, state) {
+                    if (state is CoinsLoadSuccess) {
+                      return SearchBar(state.listCoins!);
+                    }
+                    return SearchBar([]);
+                  }),
+                ],
+              ),
+            ),
+            Expanded(
+              flex: 4,
+              child: Container(
+                width: double.infinity,
+                padding: EdgeInsets.fromLTRB(20, 40, 20, 0),
+                decoration: BoxDecoration(
+                  color: ColorsApp.backgroundBottomColor,
+                  borderRadius: BorderRadius.only(
+                    topRight: Radius.circular(40),
+                    topLeft: Radius.circular(40),
+                  ),
+                ),
+                child: Column(
+                  crossAxisAlignment: CrossAxisAlignment.start,
+                  children: [
+                    Padding(
+                      padding: const EdgeInsets.symmetric(vertical: 20),
```

Comment:
we should add some variables to define the spec layout for easier change or maintenance in the future

Reply:
fixed